### PR TITLE
Allow for filtering meshctl describe results via  list of search terms

### PR DESCRIPTION
**Pull request by ryantking**
BOT NOTES: 
resolves https://github.com/solo-io/service-mesh-hub/issues/1012

### CODE REVIEW

File: pkg/meshctl/commands/describe/traffictarget/describe_traffictarget.go [hunk 1]
```diff
@@ -39,16 +39,18 @@ func Command(ctx context.Context, opts *flags.Options) *cobra.Command {
 	return cmd
 }
 
-func describeTrafficTargets(ctx context.Context, c client.Client) (string, error) {
+func describeTrafficTargets(ctx context.Context, c client.Client, searchTerms []string) (string, error) {
 	trafficTargetClient := discoveryv1alpha2.NewTrafficTargetClient(c)
-	trafficTargetList, err := trafficTargetClient.ListTrafficTarget(ctx)
+	trafficTargetList, err := trafficTargetClient.ListTrafficTarget(ctx, )
```
Comment (joekelley) [inline pkg/meshctl/commands/describe/traffictarget/describe_traffictarget.go:44]:
```suggestion
	trafficTargetList, err := trafficTargetClient.ListTrafficTarget(ctx)
```

File: pkg/meshctl/commands/describe/mesh/describe_mesh.go [hunk 2]
```diff
@@ -99,6 +101,21 @@ type meshMetadata struct {
 	Version      string
 }
 
+func matchMesh(mesh discoveryv1alpha2.Mesh, searchTerms []string) bool {
+	// do not apply matching when there are no search strings
+	if len(searchTerms) > 0 {
```
Comment (jenshu) [inline pkg/meshctl/commands/describe/mesh/describe_mesh.go:106]: should this be `== 0`? (same with the other one below)

File: pkg/meshctl/commands/describe/traffictarget/describe_traffictarget.go (hunk 1 above)
Comment (harveyxia) [inline pkg/meshctl/commands/describe/traffictarget/describe_traffictarget.go:44]:
```suggestion
	trafficTargetList, err := trafficTargetClient.ListTrafficTarget(ctx)
```

Comment (joekelley) [reply pkg/meshctl/commands/describe/traffictarget/describe_traffictarget.go:44]: nit: revert

File: pkg/meshctl/commands/describe/mesh/describe_mesh.go [hunk 3]
```diff
@@ -18,15 +18,15 @@ import (
...
+		Use:     "mesh [search terms]",
 		Short:   "Description of managed meshes",
 		Aliases: []string{"meshes"},
 		RunE: func(cmd *cobra.Command, args []string) error {
 			c, err := utils.BuildClient(opts.Kubeconfig, opts.Kubecontext)
 			if err != nil {
 				return err
 			}
-			description, err := describeMeshes(ctx, c)
+			description, err := describeMeshes(ctx, c, args)
```
Comment (harveyxia) [inline pkg/meshctl/commands/describe/mesh/describe_mesh.go:29]: We can leverage the `pflag` library's [StringSliceP() function](https://github.com/spf13/pflag/blob/master/string_slice.go#L143) for structuring the user-input search terms. This provides parsing, POSIX-style repeated CLI arguments, and also updates the command help message

File: pkg/meshctl/commands/describe/mesh/describe_mesh.go [hunk 4]
```diff
@@ -18,15 +18,15 @@ import (
 
 func Command(ctx context.Context, opts *flags.Options) *cobra.Command {
 	cmd := &cobra.Command{
-		Use:     "mesh",
+		Use:     "mesh [search terms]",
 		Short:   "Description of managed meshes",
```
Comment (harveyxia) [inline pkg/meshctl/commands/describe/mesh/describe_mesh.go:22]: Worth mentioning here that the provided terms are searched against the mesh name

**Review by harveyxia — APPROVED** [review]

**Pull request merged**